edgelet/iotedged/src/lib.rs

```diff
@@ -489,23 +493,36 @@ where
         runt_tx.send(()).unwrap_or(());
     });
 
-    core.handle().spawn(shutdown);
-
-    core.run(mgmt.join3(workload, edge_rt_with_cleanup))?;
+    let services = mgmt
+        .join3(workload, edge_rt_with_cleanup)
+        .then(|result| match result {
+            Ok(((), (), ())) => Ok(()),
+            Err(err) => {
+                error!("{}", err);
+                Err(())
+            }
+        });
+    tokio_runtime.block_on(shutdown.select(services).then(|result| match result {
```

Comment:
This `select` is wrong, since it means the services don't get to run their shutdown at all. In particular the `ModuleRuntime::stop` does not get invoked so the container is left in running state.